### PR TITLE
feat: updating shift dates banners

### PR DESCRIPTION
**Pull request by ciduarte**
**Description**
Bit of a refactor here. Took this opportunity to clean up the dates banner components after observing the existing structure was trying to do too much and further updates to it (given the updated design) would only complicate the structure more. My hope is that these components are a bit easier to read, use, and iterate on.

Also made some small updates to the timeline's styling to match the mocks.

**Dates Tab ([AA-833](https://openedx.atlassian.net/browse/AA-833))**
Suggested schedule header (verified, no past due assignments)
<div>
AA-833_verified_pastDueFalse_desktop
AA-833_verified_pastDueFalse_mobile
</div>

Shift due dates (verified, has past due assignments)
<div>
AA-833_verified_pastDueTrue_mobile
AA-833_verified_pastDueTrue_desktop
</div>

Upgrade to complete (audit, no past due assignments)
<div>
AA-833_audit_pastDueFalse_mobile
AA-833_audit_pastDueFalse_desktop
</div>

Upgrade to shift dates (audit, has past due assignments)
<div>
AA-833_audit_pastDueTrue_desktop
AA-833_audit_pastDueTrue_mobile
</div>

**Outline Tab ([AA-834](https://openedx.atlassian.net/browse/AA-834))**
<div>
AA-834_audit_pastDueTrue_desktop
AA-834_audit_pastDueTrue_mobile
</div>
<div>
AA-834_verified_pastDueTrue_mobile
AA-834_verified_pastDueTrue_desktop
</div>

### CODE REVIEW

File: src/courseware/course/sequence/sequence-navigation/SequenceNavigation.test.jsx [hunk 1]
```diff
@@ -70,7 +70,7 @@ describe('Sequence Navigation', () => {
     expect(testData.onNavigate).not.toHaveBeenCalled();
     // TODO: Not sure if this is working as expected, because the `contentType="lock"` will be overridden by the value
     //  from Redux. To make this provide a `fa-icon` lock we could introduce something like `overriddenContentType`.
-    expect(unitButton.firstChild.firstChild).toHaveClass('fa-edit');
+    expect(unitButton.firstChild).toHaveClass('fa-edit');
```
Comment (ciduarte) [inline src/courseware/course/sequence/sequence-navigation/SequenceNavigation.test.jsx:73]: I believe updating Paragon caused this change (the element is no longer so nested)

File: src/course-home/suggested-schedule-messaging/messages.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { defineMessages } from '@edx/frontend-platform/i18n';
+
+const messages = defineMessages({
+  suggestedSchedule: {
+    id: 'datesBanner.suggestedSchedule',
+    defaultMessage: 'We’ve built a suggested schedule to help you stay on track. But don’t worry—it’s flexible so you'
+      + ' can learn at your own pace.',
+  },
+  upgradeToCompleteHeader: {
+    id: 'datesBanner.upgradeToCompleteGradedBanner.header',
```
Comment (ciduarte) [inline src/course-home/suggested-schedule-messaging/messages.js:10]: Some of these strings are changing, others aren't. Left most of these ID names as is for historical purposes as to not re-trigger translation.

File: src/course-home/dates-tab/timeline/badgelist.jsx [hunk 3]
```diff
@@ -97,7 +96,7 @@ function getBadgeListAndColor(date, intl, item, items) {
           color = b.bg;
         }
         return (
-          <Badge key={b.message.id} style={marginTopStyle} className={classNames(b.bg, b.className)}>
```
Comment (ciduarte) [inline src/course-home/dates-tab/timeline/badgelist.jsx:100]: This `Badge` was a custom class. I subbed in the Paragon class (thus the removal of `Badge.jsx` and `Badge.scss`)

File: src/course-home/dates-tab/timeline/Day.jsx [hunk 4]
```diff
@@ -55,18 +55,16 @@ function Day({
...
-              /** [MM-P2P] Experiment */
-              value={mmp2pOverride ? mmp2p.state.upgradeDeadline : date}
-              day="numeric"
-              month="short"
-              weekday="short"
-              year="numeric"
-              {...timezoneFormatArgs}
-            />
-          </p>
+        <div className="row w-100 m-0 mb-1 align-items-center text-primary-700" data-testid="dates-header">
```
Comment (ciduarte) [inline src/course-home/dates-tab/timeline/Day.jsx:58]: Just some slight reformatting to align the new `<Badge />` Paragon component with the text here.

File: src/course-home/dates-tab/timeline/badgelist.jsx (hunk 3 above)
Comment (mikix) [reply src/course-home/dates-tab/timeline/badgelist.jsx:100]: Oh weird. I did that I think. Wonder why.

**Review by mikix — APPROVED** [review]
Seems fine? Looks like you broke out the one container that had all the logic for all the banners into distinct banners, to more easily control which page shows which, and compartmentalize each banner render?

File: src/course-home/dates-tab/timeline/badgelist.jsx (hunk 3 above)
Comment (ciduarte) [reply src/course-home/dates-tab/timeline/badgelist.jsx:100]: We may have only had a Pill component at the time that didn't fit our styling needs 🤷

**Pull request merged**